doc/manual/building_developing.rst

```diff
@@ -0,0 +1,172 @@
+
+In original ARTIQ, if you only intend to rebuild, without changing the source code, you may also use the ARTIQ flake's provided ``artiq-boards-shell``, a lighter environment optimized for building firmware and flashing boards. You can do this by running ``nix develop .#boards`` instead. Developers should note that in this shell the current copy of ARTIQ sources is not added to your ``PYTHONPATH``. Run ``nix flake show`` and read ``flake.nix`` carefully to understand the different available shells. 
+
+Building ARTIQ
+--------------
+
+For general troubleshooting and debugging, especially with a 'fresh' board, see also :ref:`connecting-uart`. 
+
+Kasli or KC705 (ARTIQ original) 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you have your system description JSON file on-hand, you can at this point build the firmware and gateware with a command such as: :: 
+
+    $ python -m artiq.gateware.targets.kasli <description>.json
+    $ python -m artiq.gateware.targets.kc705 <description>.json
+
+This will create a directory ``artiq_kasli`` or ``artiq_kc705`` containing the binaries in a subdirectory named after your variant. Flash the board as described in :ref:`writing-flash`, plus the option ``--srcbuild``, e.g., if your board is correctly connected by JTAG USB: :: 
+
+    $ artiq_flash --srcbuild [-t kc705] -d artiq_<board>/<variant> 
+
+Kasli-SoC or ZC706 (ARTIQ on Zynq)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The building process for Zynq devices is a little more complex. The easiest method for Kasli-SoC is to leverage ``nix build`` in the provided flake. Depending on your targeted DRTIO role, replace one of ``demo.json`` (i.e. standalone), ``kasli-soc-master.json``, or ``kasli-soc-satellite.json`` with your own system description, 
+renaming it to match, and in the root of the repository run the corresponding version of the build command: ::  
+
+    $ nix build .#kasli_soc-demo-sd 
+    $ nix build .#kasli_soc-master-sd 
+    $ nix build .#kasli_soc-satellite-sd 
+
+This should produce a directory ``result`` which contains a file ``boot.bin``. As described in :ref:`writing-flash`, if your Kasli-SoC is currently accessible over the network, it can be flashed with ``artiq_coremgmt``. If it is not connected to the network:
+
+1. Power off the board, extract the SD card and load ``boot.bin`` onto it manually. 
+2. Insert the SD card back into the board. 
+3. Ensure that the DIP switches (labeled BOOT MODE) are set correctly, to SD.
+4. Power the board back on. 
+
+Optionally, the SD card may also be loaded at the same time with an additional file ``config.txt``, which can contain preset configuration values in the format ``key=value``, one per line. The keys are those used with ``artiq_coremgmt``. This allows e.g. presetting an IP address and other networking information. 
```

Comment:
IP address is pretty much all of the networking configuration, but there are other keys involved, like rtio_clock, sed spreading, not mentioning routing table bin file...

Reply:
e.g. means "for example", so I don't think the full list has to be run through here -- everything else except IP is pretty easily set after the fact with artiq_coremgmt, IP is the only thing you really benefit from having in advance.  And there's like five keys for networking if you count both IPs, both gateways, MAC address...